### PR TITLE
chore(datasets): Remove deprecation warning from Table Dataset

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Remove deprecation warning as suggested [here](https://github.com/kedro-org/kedro-plugins/issues/942) -  Need to remove code that's marked as "to be removed": ibis dataset: https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/kedro_datasets/ibis/table_dataset.py#L123-L128

## Development notes

* Removed the deprecation warning from Table Dataset
* Updated release note

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes
- [ ] Received approvals from at least half of the TSC (required for adding a new, non-experimental dataset)

### CODE REVIEW

File: kedro-datasets/RELEASE.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 ## Bug fixes and other changes
 - Implemented Snowflake's (local testing framework)[https://docs.snowflake.com/en/developer-guide/snowpark/python/testing-locally] for testing purposes
 - Improved the dependency management for Spark-based datasets by refactoring the Spark and Databricks utility functions used across the datasets.
+- Remove deprecation warning to load and save files with Ibis backend from Table Dataset.
```
Comment (merelcht) [inline kedro-datasets/RELEASE.md:17]: You don't need to add this to the release notes.

**Review by merelcht — COMMENTED** [review]
@deepyaman should `filepath` and `file_format` also be removed as arguments? I see in the history that you did remove them from the docstrings already.

**Comment by deepyaman** [comment]
> @deepyaman should `filepath` and `file_format` also be removed as arguments? I see in the history that you did remove them from the docstrings already.

Yes, please!

**Comment by ravi-kumar-pilla** [comment]
> Yes, please!

Hi @deepyaman , I may not be fully aware of the context but what happens to the [load](https://github.com/kedro-org/kedro-plugins/pull/956/files#diff-44ba1ac4887f3718c12bb8156188ad2cc642baf112ff42e555eaea707e36f139R160) function if we remove the parameters `filepath` and `file_format` ?  I see a docstring mentioning [this](https://github.com/kedro-org/kedro-plugins/pull/956/files#diff-44ba1ac4887f3718c12bb8156188ad2cc642baf112ff42e555eaea707e36f139R94) 

So to summarize, this will affect 2 functions `load` and `_describe`. For `_describe` I can remove the keys from the returned dict. I am not clear on `load`

Thank you

**Comment by deepyaman** [comment]
> > Yes, please!
> 
> Hi @deepyaman , I may not be fully aware of the context but what happens to the [load](https://github.com/kedro-org/kedro-plugins/pull/956/files#diff-44ba1ac4887f3718c12bb8156188ad2cc642baf112ff42e555eaea707e36f139R160) function if we remove the parameters `filepath` and `file_format` ? I see a docstring mentioning [this](https://github.com/kedro-org/kedro-plugins/pull/956/files#diff-44ba1ac4887f3718c12bb8156188ad2cc642baf112ff42e555eaea707e36f139R94)
> 
> So to summarize, this will affect 2 functions `load` and `_describe`. For `_describe` I can remove the keys from the returned dict. I am not clear on `load`
> 
> Thank you

All this logic can be removed; `ibis.FileDataset` should fully cover these functionalities.

```python
        if self._filepath is not None:
            if self._file_format is None:
                raise NotImplementedError
            reader = getattr(self.connection, f"read_{self._file_format}")
            return reader(self._filepath, self._table_name, **self._load_args)

        else:
```

File: kedro-datasets/kedro_datasets/ibis/table_dataset.py [hunk 2]
```diff
@@ -115,31 +107,13 @@ def __init__(  # noqa: PLR0913
...
-                "Dataset' in Kedro-Datasets 6.0.0",
-                KedroDeprecationWarning,
-                stacklevel=2,
-            )
-
-        self._filepath = filepath
-        self._file_format = file_format
+        if table_name is None:
+            raise DatasetError("Must provide `table_name`.")
+
```
Comment (deepyaman) [inline kedro-datasets/kedro_datasets/ibis/table_dataset.py:112]:
```suggestion
        if table_name is None:
            raise DatasetError("Must provide `table_name`.")

```
Not necessary; can just make the argument a required, positional argument.

File: kedro-datasets/kedro_datasets/ibis/table_dataset.py [hunk 3]
```diff
@@ -159,29 +133,16 @@ def connection(self) -> BaseBackend:
...
         writer = getattr(self.connection, f"create_{self._materialized}")
         writer(self._table_name, data, **self._save_args)
 
     def _describe(self) -> dict[str, Any]:
         return {
-            "filepath": self._filepath,
-            "file_format": self._file_format,
             "table_name": self._table_name,
             "backend": self._connection_config["backend"],
-            "load_args": self._load_args,
```
Comment (deepyaman) [inline kedro-datasets/kedro_datasets/ibis/table_dataset.py:184]: This shouldn't be removed.

File: kedro-datasets/tests/ibis/test_table_dataset.py [hunk 4]
```diff
@@ -103,16 +79,11 @@ def test_save_extra_params(self, table_dataset, save_args, dummy_table, database
         )
         assert not con.sql("SELECT * FROM duckdb_views").fetchnumpy()["view_name"]
 
-    def test_no_filepath_or_table_name(connection_config):
-        pattern = r"Must provide at least one of `filepath` or `table_name`\."
+    def test_no_table_name(connection_config):
+        pattern = r"Must provide `table_name`\."
```
Comment (deepyaman) [inline kedro-datasets/tests/ibis/test_table_dataset.py:83]: Can just remove this test, if the argument is required.

**Review by deepyaman — CHANGES_REQUESTED** [review]
Nevermind, I reviewed it.

File: kedro-datasets/kedro_datasets/ibis/table_dataset.py (hunk 3 above)
Comment (ravi-kumar-pilla) [reply kedro-datasets/kedro_datasets/ibis/table_dataset.py:184]: We are not using load_args anywhere, should you still need this passed to the init ?

File: kedro-datasets/kedro_datasets/ibis/table_dataset.py [hunk 5]
```diff
@@ -1,14 +1,13 @@
 """Provide data loading and saving functionality for Ibis's backends."""
+
```
Comment (deepyaman) [inline kedro-datasets/kedro_datasets/ibis/table_dataset.py:2]:
```suggestion
```
Nit: Just for consistency/minimal changes.

File: kedro-datasets/kedro_datasets/ibis/table_dataset.py [hunk 6]
```diff
@@ -159,26 +140,14 @@ def connection(self) -> BaseBackend:
...
     def load(self) -> ir.Table:
-        if self._filepath is not None:
-            if self._file_format is None:
-                raise NotImplementedError
-
-            reader = getattr(self.connection, f"read_{self._file_format}")
-            return reader(self._filepath, self._table_name, **self._load_args)
-        else:
-            return self.connection.table(self._table_name)
+        return self.connection.table(self._table_name)
```
Comment (deepyaman) [inline kedro-datasets/kedro_datasets/ibis/table_dataset.py:143]: Just noting that this will be (slightly) different after #909.

File: kedro-datasets/tests/ibis/test_table_dataset.py [hunk 7]
```diff
@@ -86,11 +74,6 @@ def test_exists(self, table_dataset, dummy_table):
         table_dataset.save(dummy_table)
         assert table_dataset.exists()
 
-    @pytest.mark.parametrize("load_args", [{"filename": True}], indirect=True)
```
Comment (deepyaman) [inline kedro-datasets/tests/ibis/test_table_dataset.py:89]: I think this test should be kept, but the `load_args` passed can be something different, like `{"database": "test"}` (after #909).

**Review by deepyaman — COMMENTED** [review]
This largely looks good to me!

One more change that probably should be added is including the removal (and directions to use `FileDataset` instead) in the release notes.

File: kedro-datasets/tests/ibis/test_table_dataset.py (hunk 7 above)
Comment (ravi-kumar-pilla) [reply kedro-datasets/tests/ibis/test_table_dataset.py:115]: Hi @deepyaman , I tried adding below test, but it keeps failing saying `Catalog "random" does not exist!`. I am not sure what I am missing. Could you please help me in resolving the issue and suggest what else I can assert for the `load_args` test. Thank you

```
@pytest.mark.parametrize("load_args", [{"database": "random.db"}], indirect=True)
    def test_load_extra_params(
        self, dummy_table, tmp_path, load_args
    ):
        """Test overriding the load arguments."""
        db_path = tmp_path / load_args["database"]
        table_dataset = TableDataset(table_name="test", load_args=load_args)
        con = duckdb.connect(str(db_path))
        table_dataset.save(dummy_table)
        table_dataset.load()
     
        assert (
            "test" in con.sql("SELECT * FROM duckdb_views").fetchnumpy()["schema_name"]
        )
```

File: kedro-datasets/tests/ibis/test_table_dataset.py [hunk 8]
```diff
@@ -112,10 +100,10 @@ def test_exists(self, table_dataset, dummy_table):
...
-    def test_load_extra_params(self, table_dataset_from_csv, load_args):
-        """Test overriding the default load arguments."""
-        assert "filename" in table_dataset_from_csv.load()
+    def test_load_extra_params(self):
+        """Test overriding the load arguments."""
+        table_dataset = TableDataset(table_name="test", load_args={"database": "test"})
+        assert "database" in table_dataset._load_args
```
Comment (deepyaman) [inline kedro-datasets/tests/ibis/test_table_dataset.py:106]:
```suggestion
    @pytest.mark.parametrize("load_args", [{"filename": True}], indirect=True)
    def test_load_extra_params(self, table_dataset, load_args):
        """Test overriding the default load arguments."""
        assert table_dataset._load_args["database"] == "test"
```

or even

```suggestion
    @pytest.mark.parametrize("load_args", [{"filename": True}], indirect=True)
    def test_load_extra_params(self, table_dataset, load_args):
        """Test overriding the default load arguments."""
        for key, value in load_args.items()
            assert table_dataset._load_args[key] == value
```

if you want to be extra consistent with the other dataset tests

**Review by deepyaman — APPROVED** [review]
Approving, only minor suggestions

**Review by ankatiyar — APPROVED** [review]
LGTM!
